shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -317,6 +318,29 @@ - (NSOpenGLContext*)resourceContext {
   return _resourceContext;
 }
 
+- (float)displayRefreshRate {
+  NSDictionary* description = [[NSScreen mainScreen] deviceDescription];
```

Comment:
Why the main screen? What if the Flutter view is on a different screen?

It seems like the proposed embedder API should be reconsidered here to be more flexible for desktop use cases. (E.g., what happens when we support multiple windows, and there are windows on different screens?)

Reply:
[`mainScreen`](https://developer.apple.com/documentation/appkit/nsscreen/1388371-mainscreen?language=objc) is defined as:

> The main screen is not necessarily the same screen that contains the menu bar or has its origin at (0, 0). The main screen refers to the screen containing the window that is currently receiving keyboard events. It is the main screen because it is the one with which the user is most likely interacting.

Given that we currently only support 1 window the above is most likely what the devtools wants to display.

In the cases where we support multiple windows, I think we would want the embedder to be able to answer the following questions IMO:

1. What are all the available displays and their refresh rates?
2. What displays do each of my windows span?
3. What configurations/mode are the displays in? (Mirroring for example could limit the refresh rate of the display being mirrored to).

@stuartmorgan are there any other things you think I should consider for the embedder API.

An open question I have is, given that currently devtools and other benchmarking infra only supports configurations with 1 display, I'm wondering if we should treat displays with the most number of windows in them as the main display and use that while we add support for multiple displays and revise this assumption.

Reply:
> Given that we currently only support 1 window

For now; we intend to change that, and should not design new APIs (in a surface where we have a firm policy against any breaking changes) that aren't compatible with that plan.

> What displays do each of my windows span?

We'll need that API for multi-window support in general, not just refresh rate, so I think we can assume that will exist.

> What configurations/mode are the displays in? (Mirroring for example could limit the refresh rate of the display being mirrored to).

Wouldn't that be reflected in the answer to `1`? I.e., wouldn't that just be a singe display with the limed refresh rate?

Reply:
> Wouldn't that be reflected in the answer to 1? I.e., wouldn't that just be a singe display with the limed refresh rate?

I was thinking of this as a static configuration we pass during engine initialization. On reconsideration, I think it makes sense for this to be a callback that the embedder provides to account for dynamic nature of displays. I will update the PR: https://github.com/flutter/engine/pull/21094 to account for these changes.